### ISSUE TITLE
Please check security email immediately

### ISSUE THREAD

**Issue by benmccann** [typescript-eslint/typescript-eslint#10955]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [x] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

n/a

### Repro Code

```TypeScript
Sent in email. I wanted to file this just in case that email wasn't super actively monitored. Sorry if this is extra noise. Please feel free to close if you've received the email
```

### ESLint Config

```javascript
n/a
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    // ...
  }
}
```

### Expected Result

n/a

### Actual Result

n/a
